gdk4/src/subclass/content_provider.rs

```diff
@@ -0,0 +1,248 @@
+}
+
+impl<T: ContentProviderImpl> ContentProviderImplExt for T {
+    fn parent_content_changed(&self, provider: &Self::Type) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GdkContentProviderClass;
+            if let Some(f) = (*parent_class).content_changed {
+                f(provider
+                    .unsafe_cast_ref::<ContentProvider>()
+                    .to_glib_none()
+                    .0)
+            }
+        }
+    }
+
+    fn parent_attach_clipboard(&self, provider: &Self::Type, clipboard: &Clipboard) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GdkContentProviderClass;
+            if let Some(f) = (*parent_class).attach_clipboard {
+                f(
+                    provider
+                        .unsafe_cast_ref::<ContentProvider>()
+                        .to_glib_none()
+                        .0,
+                    clipboard.to_glib_none().0,
+                )
+            }
+        }
+    }
+
+    fn parent_detach_clipboard(&self, provider: &Self::Type, clipboard: &Clipboard) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GdkContentProviderClass;
+            if let Some(f) = (*parent_class).detach_clipboard {
+                f(
+                    provider
+                        .unsafe_cast_ref::<ContentProvider>()
+                        .to_glib_none()
+                        .0,
+                    clipboard.to_glib_none().0,
+                )
+            }
+        }
+    }
+
+    fn parent_ref_formats(&self, provider: &Self::Type) -> ContentFormats {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GdkContentProviderClass;
+            let f = (*parent_class)
+                .ref_formats
+                .expect("no parent \"ref_formats\" implementation");
+            let ret = f(provider
+                .unsafe_cast_ref::<ContentProvider>()
+                .to_glib_none()
+                .0);
+
+            from_glib_full(ret)
+        }
+    }
+
+    fn parent_ref_storable_formats(&self, provider: &Self::Type) -> ContentFormats {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().parent_class() as *mut ffi::GdkContentProviderClass;
+            let f = (*parent_class)
+                .ref_storable_formats
+                .expect("no parent \"ref_storable_formats\" implementation");
+            let ret = f(provider
+                .unsafe_cast_ref::<ContentProvider>()
+                .to_glib_none()
+                .0);
+
+            from_glib_full(ret)
+        }
+    }
+
+    fn parent_value(&self, provider: &Self::Type, value: &mut Value) -> Result<(), glib::Error> {
```

Comment:
```suggestion
    fn parent_value(&self, provider: &Self::Type) -> Result<Value, glib::Error> {
```
Or the user has to initialize the value with a specific type before?

Reply:
Yes, the value is pre-initialized with the requested type.